### DISCUSSION TITLE
How to update a field from a budibase DB

### DISCUSSION THREAD

**Discussion by jaarroyl** [Budibase/budibase#9414] [Help] [answered]
Hi guys, I want to update a field using the following form:

![Captura de pantalla 2023-01-23 a la(s) 12 13 01](https://user-images.githubusercontent.com/41343847/214076180-218b314a-e075-4366-a136-3ab7b8d55791.png)

In the form Im using a "On change" action that allow to recognize if the written "certificacion" exist in the database.
![image](https://user-images.githubusercontent.com/41343847/214076579-63595119-8335-4579-85a0-68cd5bd4e358.png)

If there exist, I want to be able one clicking on "Crear/Modificar certificacion" to modify the "Lugar instancia" value in the dabatase.
![image](https://user-images.githubusercontent.com/41343847/214076987-c6678614-3693-4f39-abb8-a9d7a08f80e1.png)

I know that I could use the _id value of the row in the database. The problem is that I dont know how to get the value according to the "certificacion" field written in the form.

Thanks

**Comment by melohagan** [comment, marked as answer]
Hey @jaarroyl 

Have you had a look at this answer? https://github.com/Budibase/budibase/discussions/6294

**Reply by jaarroyl** [reply]
Hey @melohagan, that answer doesnt solve my problem. What I want is to update a specific row in budibase table according to the value filled into the form field. I dont know how to get the _id value of the row that I need to update.

Best,

**Reply by melohagan** [reply]
If I understand correctly, the process will be fairly similar. You'll want a data provider with filter to get the row(s) that match the user input.

Then instead of doing validation, you would perform your **Save Row** action using the output of the filtered data provider.

Something like:

> Form 
> > Certification field
> > Data Provider (filter on certification field value)
> > > Button (submit form. Save row id with binding: `{{ Data provider.rows.0._id }}`

But you probably would also want validation, because what if the user input does not match an existing row?

**Reply by jaarroyl** [reply]
Hi @melohagan, 

I did what you reply.

I inserted a new data provider that allow us to filter according the selection of the "certification" field.

<img width="205" alt="image" src="https://user-images.githubusercontent.com/41343847/214824668-b9244042-e7b5-4952-b9f1-1e9e348de2dc.png">

{{ manage_instances.Fields.certificacion }}
<img width="855" alt="image" src="https://user-images.githubusercontent.com/41343847/214824853-82c4ac6d-b68a-4ba1-ba52-289ce109908c.png">

Then in the "saved button" I write the binding using the filter Rows.
{{ filter.Rows.0._id }}

<img width="523" alt="image" src="https://user-images.githubusercontent.com/41343847/214825126-0572b872-11b2-48fb-b750-52afd9d3c10b.png">

Ones I tried, 

This is the current data in the table:

<img width="1116" alt="image" src="https://user-images.githubusercontent.com/41343847/214825353-755fbd2d-2209-4135-a8e0-b0c4dc224acb.png">

Suppose that I want to modify the "Periodo de descanso" row.

<img width="1302" alt="image" src="https://user-images.githubusercontent.com/41343847/214825503-eb2fa76c-d28e-4560-93d1-aa664bac30de.png">

Ones I wrote "Periodo de descanso" I have a Change Button that update the field "Lugar" if the value of the field "Certificacion" exists in the DB.

Imagine that I want to update the Lugar to "baño" instead of "sala".

Ones I tried and click Save.

<img width="1306" alt="image" src="https://user-images.githubusercontent.com/41343847/214825821-fe2b5939-cb9b-4ab2-bd34-b8a2e24c1d4b.png">

The "Periodo de descanso" is duplicated so I assume that is not getting the _id correctly to make an update.

<img width="1123" alt="image" src="https://user-images.githubusercontent.com/41343847/214825996-e41ca416-ca0c-4998-b80c-900958d7f8bf.png">

What I did wrong?

Best,

**Reply by melohagan** [reply]
This needs to be set to **Binding** in the dropdown:
![image](https://user-images.githubusercontent.com/101575380/214826631-840ec9b3-8c37-4ee8-a615-945f7b404a2b.png)

**Reply by jaarroyl** [reply]
Thanks a lot for the solution @melohagan 

Now is working.

And I decided to use Javascript to validate if the "certificacion" exists before update.

If it does, it will get the _id value from the filter and if not will let the _id value in blank saving a new "certificacion" to de DB.

`const idSelected = $("filter.Rows").find(r => r["certificacion"] === $("manage_instances.Fields.certificacion"));

return idSelected ? idSelected._id : "";`

One of the things that I think budibase must have, is the State tree like Retool has. For people that doesnt know a lot of programing make easier to use the Key values of the dict.

<img width="284" alt="image" src="https://user-images.githubusercontent.com/41343847/214848002-30bcd3bf-9a64-43c6-9808-e2cc0798d309.png">

Best for you!!
